src/Particular.Operations.Heartbeats.Api/IProcessHeartbeats.cs

```diff
@@ -1,8 +1,10 @@
 ﻿namespace Particular.Operations.Heartbeats.Api
 {
+    using System.Threading.Tasks;
+
     public interface IProcessHeartbeats
     {
-        void Handle(RegisterEndpointStartup endpointStartup);
-        void Handle(EndpointHeartbeat heartbeat);
+        Task Handle(RegisterEndpointStartup endpointStartup);
```

Comment:
Doesn't this mean that every host will have to know how to listen for `EndpointHeartbeat` and `RegisterEndpointStartup` commands? Should these handlers live within this component?

Reply:
@mikeminutillo these would live in the `Heartbeat engine` so neither host nor component. In this case the `Uptime` component requires this API. If a host/particle has this engine enabled it exposes the API and the component can be initialized.

In the current phase SC contains both the host code and the engines :-(